[2024-01-07T19:31:13.874+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: ingest_from_postgres_to_GCS_bucket.course_load_dataset_to_bq scheduled__2024-01-06T00:00:00+00:00 [queued]>
[2024-01-07T19:31:13.901+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: ingest_from_postgres_to_GCS_bucket.course_load_dataset_to_bq scheduled__2024-01-06T00:00:00+00:00 [queued]>
[2024-01-07T19:31:13.902+0000] {taskinstance.py:1361} INFO - Starting attempt 1 of 3
[2024-01-07T19:31:13.973+0000] {taskinstance.py:1382} INFO - Executing <Task(GCSToBigQueryOperator): course_load_dataset_to_bq> on 2024-01-06 00:00:00+00:00
[2024-01-07T19:31:13.982+0000] {standard_task_runner.py:57} INFO - Started process 9978 to run task
[2024-01-07T19:31:13.985+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'ingest_from_postgres_to_GCS_bucket', 'course_load_dataset_to_bq', 'scheduled__2024-01-06T00:00:00+00:00', '--job-id', '448', '--raw', '--subdir', 'DAGS_FOLDER/postgres_to_gcs_staging.py', '--cfg-path', '/tmp/tmpfpouo4z1']
[2024-01-07T19:31:13.989+0000] {standard_task_runner.py:85} INFO - Job 448: Subtask course_load_dataset_to_bq
[2024-01-07T19:31:14.130+0000] {task_command.py:416} INFO - Running <TaskInstance: ingest_from_postgres_to_GCS_bucket.course_load_dataset_to_bq scheduled__2024-01-06T00:00:00+00:00 [running]> on host 5d007c0b4985
[2024-01-07T19:31:14.375+0000] {taskinstance.py:1662} INFO - Exporting env vars: AIRFLOW_CTX_DAG_EMAIL='franklinmutwiri41@gmail.com' AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='ingest_from_postgres_to_GCS_bucket' AIRFLOW_CTX_TASK_ID='course_load_dataset_to_bq' AIRFLOW_CTX_EXECUTION_DATE='2024-01-06T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-01-06T00:00:00+00:00'
[2024-01-07T19:31:14.393+0000] {connection.py:232} WARNING - Connection schemes (type: google_cloud_platform) shall not contain '_' according to RFC3986.
[2024-01-07T19:31:14.399+0000] {base.py:73} INFO - Using connection ID 'google_cloud_default' for task execution.
[2024-01-07T19:31:14.400+0000] {gcs_to_bigquery.py:376} INFO - Using existing BigQuery table for storing data...
[2024-01-07T19:31:14.451+0000] {gcs_to_bigquery.py:380} INFO - Executing: {'load': {'autodetect': True, 'createDisposition': 'CREATE_NEVER', 'destinationTable': {'projectId': 'data-warehousing-proj', 'datasetId': 'staging_school', 'tableId': 'STG_COURSE'}, 'sourceFormat': 'PARQUET', 'sourceUris': ['gs://school_data_lake_data-warehousing-proj/data/course-20240106.parquet'], 'writeDisposition': 'WRITE_TRUNCATE', 'ignoreUnknownValues': False}}
[2024-01-07T19:31:14.452+0000] {bigquery.py:1596} INFO - Inserting job ***_ingest_from_postgres_to_GCS_bucket_course_load_dataset_to_bq_2024_01_06T00_00_00_00_00_ba2c52dedbbbec243c12c05cd1b0b4db
[2024-01-07T19:31:18.495+0000] {taskinstance.py:1937} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/google/cloud/transfers/gcs_to_bigquery.py", line 439, in execute
    job.result(timeout=self.result_timeout, retry=self.result_retry)
  File "/home/airflow/.local/lib/python3.8/site-packages/google/cloud/bigquery/job/base.py", line 922, in result
    return super(_AsyncJob, self).result(timeout=timeout, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/google/api_core/future/polling.py", line 261, in result
    raise self._exception
google.api_core.exceptions.NotFound: 404 Not found: URI gs://school_data_lake_data-warehousing-proj/data/course-20240106.parquet
[2024-01-07T19:31:18.519+0000] {taskinstance.py:1400} INFO - Marking task as UP_FOR_RETRY. dag_id=ingest_from_postgres_to_GCS_bucket, task_id=course_load_dataset_to_bq, execution_date=20240106T000000, start_date=20240107T193113, end_date=20240107T193118
[2024-01-07T19:31:18.559+0000] {logging_mixin.py:151} WARNING - /home/***/.local/lib/python3.8/site-packages/***/utils/email.py:154 RemovedInAirflow3Warning: Fetching SMTP credentials from configuration variables will be deprecated in a future release. Please set credentials using a connection instead.
[2024-01-07T19:31:18.560+0000] {configuration.py:1068} WARNING - section/key [smtp/smtp_user] not found in config
[2024-01-07T19:31:18.560+0000] {email.py:270} INFO - Email alerting: attempt 1
[2024-01-07T19:31:18.576+0000] {configuration.py:1068} WARNING - section/key [smtp/smtp_user] not found in config
[2024-01-07T19:31:18.577+0000] {email.py:270} INFO - Email alerting: attempt 1
[2024-01-07T19:31:18.578+0000] {taskinstance.py:2007} ERROR - Failed to send email to: ['franklinmutwiri41@gmail.com']
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 1518, in _run_raw_task
    self._execute_task_with_callbacks(context, test_mode, session=session)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 1681, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 1744, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/google/cloud/transfers/gcs_to_bigquery.py", line 439, in execute
    job.result(timeout=self.result_timeout, retry=self.result_retry)
  File "/home/airflow/.local/lib/python3.8/site-packages/google/cloud/bigquery/job/base.py", line 922, in result
    return super(_AsyncJob, self).result(timeout=timeout, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/google/api_core/future/polling.py", line 261, in result
    raise self._exception
google.api_core.exceptions.NotFound: 404 Not found: URI gs://school_data_lake_data-warehousing-proj/data/course-20240106.parquet

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 2420, in email_alert
    send_email(task.email, subject, html_content)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/email.py", line 80, in send_email
    return backend(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/email.py", line 154, in send_email_smtp
    send_mime_email(e_from=mail_from, e_to=recipients, mime_msg=msg, conn_id=conn_id, dryrun=dryrun)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/email.py", line 272, in send_mime_email
    smtp_conn = _get_smtp_connection(smtp_host, smtp_port, smtp_timeout, smtp_ssl)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/email.py", line 316, in _get_smtp_connection
    return smtplib.SMTP(host=host, port=port, timeout=timeout)
  File "/usr/local/lib/python3.8/smtplib.py", line 255, in __init__
    (code, msg) = self.connect(host, port)
  File "/usr/local/lib/python3.8/smtplib.py", line 339, in connect
    self.sock = self._get_socket(host, port, self.timeout)
  File "/usr/local/lib/python3.8/smtplib.py", line 310, in _get_socket
    return socket.create_connection((host, port), timeout,
  File "/usr/local/lib/python3.8/socket.py", line 808, in create_connection
    raise err
  File "/usr/local/lib/python3.8/socket.py", line 796, in create_connection
    sock.connect(sa)
OSError: [Errno 99] Cannot assign requested address

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 2005, in handle_failure
    self.email_alert(error, task)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 2422, in email_alert
    send_email(task.email, subject, html_content_err)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/email.py", line 80, in send_email
    return backend(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/email.py", line 154, in send_email_smtp
    send_mime_email(e_from=mail_from, e_to=recipients, mime_msg=msg, conn_id=conn_id, dryrun=dryrun)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/email.py", line 272, in send_mime_email
    smtp_conn = _get_smtp_connection(smtp_host, smtp_port, smtp_timeout, smtp_ssl)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/email.py", line 316, in _get_smtp_connection
    return smtplib.SMTP(host=host, port=port, timeout=timeout)
  File "/usr/local/lib/python3.8/smtplib.py", line 255, in __init__
    (code, msg) = self.connect(host, port)
  File "/usr/local/lib/python3.8/smtplib.py", line 339, in connect
    self.sock = self._get_socket(host, port, self.timeout)
  File "/usr/local/lib/python3.8/smtplib.py", line 310, in _get_socket
    return socket.create_connection((host, port), timeout,
  File "/usr/local/lib/python3.8/socket.py", line 808, in create_connection
    raise err
  File "/usr/local/lib/python3.8/socket.py", line 796, in create_connection
    sock.connect(sa)
OSError: [Errno 99] Cannot assign requested address
[2024-01-07T19:31:18.617+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 448 for task course_load_dataset_to_bq (404 Not found: URI gs://school_data_lake_data-warehousing-proj/data/course-20240106.parquet; 9978)
[2024-01-07T19:31:18.662+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2024-01-07T19:31:18.697+0000] {taskinstance.py:2778} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-01-07T19:59:07.985+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: ingest_from_postgres_to_GCS_bucket.course_load_dataset_to_bq scheduled__2024-01-06T00:00:00+00:00 [queued]>
[2024-01-07T19:59:08.018+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: ingest_from_postgres_to_GCS_bucket.course_load_dataset_to_bq scheduled__2024-01-06T00:00:00+00:00 [queued]>
[2024-01-07T19:59:08.018+0000] {taskinstance.py:1361} INFO - Starting attempt 1 of 3
[2024-01-07T19:59:08.158+0000] {taskinstance.py:1382} INFO - Executing <Task(GCSToBigQueryOperator): course_load_dataset_to_bq> on 2024-01-06 00:00:00+00:00
[2024-01-07T19:59:08.170+0000] {standard_task_runner.py:57} INFO - Started process 10709 to run task
[2024-01-07T19:59:08.178+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'ingest_from_postgres_to_GCS_bucket', 'course_load_dataset_to_bq', 'scheduled__2024-01-06T00:00:00+00:00', '--job-id', '462', '--raw', '--subdir', 'DAGS_FOLDER/postgres_to_gcs_staging.py', '--cfg-path', '/tmp/tmpl1ynxdc3']
[2024-01-07T19:59:08.182+0000] {standard_task_runner.py:85} INFO - Job 462: Subtask course_load_dataset_to_bq
[2024-01-07T19:59:08.283+0000] {task_command.py:416} INFO - Running <TaskInstance: ingest_from_postgres_to_GCS_bucket.course_load_dataset_to_bq scheduled__2024-01-06T00:00:00+00:00 [running]> on host 5d007c0b4985
[2024-01-07T19:59:08.467+0000] {taskinstance.py:1662} INFO - Exporting env vars: AIRFLOW_CTX_DAG_EMAIL='franklinmutwiri41@gmail.com' AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='ingest_from_postgres_to_GCS_bucket' AIRFLOW_CTX_TASK_ID='course_load_dataset_to_bq' AIRFLOW_CTX_EXECUTION_DATE='2024-01-06T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-01-06T00:00:00+00:00'
[2024-01-07T19:59:08.496+0000] {connection.py:232} WARNING - Connection schemes (type: google_cloud_platform) shall not contain '_' according to RFC3986.
[2024-01-07T19:59:08.506+0000] {base.py:73} INFO - Using connection ID 'google_cloud_default' for task execution.
[2024-01-07T19:59:08.510+0000] {gcs_to_bigquery.py:376} INFO - Using existing BigQuery table for storing data...
[2024-01-07T19:59:08.593+0000] {gcs_to_bigquery.py:380} INFO - Executing: {'load': {'autodetect': True, 'createDisposition': 'CREATE_NEVER', 'destinationTable': {'projectId': 'data-warehousing-proj', 'datasetId': 'staging_school', 'tableId': 'STG_COURSE'}, 'sourceFormat': 'PARQUET', 'sourceUris': ['gs://school_data_lake_data-warehousing-proj/data/course.parquet'], 'writeDisposition': 'WRITE_TRUNCATE', 'ignoreUnknownValues': False}}
[2024-01-07T19:59:08.594+0000] {bigquery.py:1596} INFO - Inserting job ***_ingest_from_postgres_to_GCS_bucket_course_load_dataset_to_bq_2024_01_06T00_00_00_00_00_0f20cf8ca204c2e5989275beb969aff7
[2024-01-07T19:59:11.627+0000] {taskinstance.py:1400} INFO - Marking task as SUCCESS. dag_id=ingest_from_postgres_to_GCS_bucket, task_id=course_load_dataset_to_bq, execution_date=20240106T000000, start_date=20240107T195907, end_date=20240107T195911
[2024-01-07T19:59:11.683+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 0
[2024-01-07T19:59:11.712+0000] {taskinstance.py:2778} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-01-07T22:07:12.881+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: ingest_from_postgres_to_GCS_bucket.course_load_dataset_to_bq scheduled__2024-01-06T00:00:00+00:00 [queued]>
[2024-01-07T22:07:12.921+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: ingest_from_postgres_to_GCS_bucket.course_load_dataset_to_bq scheduled__2024-01-06T00:00:00+00:00 [queued]>
[2024-01-07T22:07:12.922+0000] {taskinstance.py:1361} INFO - Starting attempt 1 of 3
[2024-01-07T22:07:12.981+0000] {taskinstance.py:1382} INFO - Executing <Task(GCSToBigQueryOperator): course_load_dataset_to_bq> on 2024-01-06 00:00:00+00:00
[2024-01-07T22:07:12.989+0000] {standard_task_runner.py:57} INFO - Started process 1471 to run task
[2024-01-07T22:07:12.997+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'ingest_from_postgres_to_GCS_bucket', 'course_load_dataset_to_bq', 'scheduled__2024-01-06T00:00:00+00:00', '--job-id', '556', '--raw', '--subdir', 'DAGS_FOLDER/postgres_to_gcs_staging.py', '--cfg-path', '/tmp/tmp7i44klxk']
[2024-01-07T22:07:13.000+0000] {standard_task_runner.py:85} INFO - Job 556: Subtask course_load_dataset_to_bq
[2024-01-07T22:07:13.098+0000] {task_command.py:416} INFO - Running <TaskInstance: ingest_from_postgres_to_GCS_bucket.course_load_dataset_to_bq scheduled__2024-01-06T00:00:00+00:00 [running]> on host 6609460de6e5
[2024-01-07T22:07:13.252+0000] {taskinstance.py:1662} INFO - Exporting env vars: AIRFLOW_CTX_DAG_EMAIL='franklinmutwiri41@gmail.com' AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='ingest_from_postgres_to_GCS_bucket' AIRFLOW_CTX_TASK_ID='course_load_dataset_to_bq' AIRFLOW_CTX_EXECUTION_DATE='2024-01-06T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-01-06T00:00:00+00:00'
[2024-01-07T22:07:13.268+0000] {connection.py:232} WARNING - Connection schemes (type: google_cloud_platform) shall not contain '_' according to RFC3986.
[2024-01-07T22:07:13.273+0000] {base.py:73} INFO - Using connection ID 'google_cloud_default' for task execution.
[2024-01-07T22:07:13.274+0000] {gcs_to_bigquery.py:376} INFO - Using existing BigQuery table for storing data...
[2024-01-07T22:07:13.367+0000] {gcs_to_bigquery.py:380} INFO - Executing: {'load': {'autodetect': True, 'createDisposition': 'CREATE_NEVER', 'destinationTable': {'projectId': 'data-warehousing-proj', 'datasetId': 'staging_school', 'tableId': 'STG_COURSE'}, 'sourceFormat': 'PARQUET', 'sourceUris': ['gs://school_data_lake_data-warehousing-proj/data/course.parquet'], 'writeDisposition': 'WRITE_TRUNCATE', 'ignoreUnknownValues': False}}
[2024-01-07T22:07:13.368+0000] {bigquery.py:1596} INFO - Inserting job ***_ingest_from_postgres_to_GCS_bucket_course_load_dataset_to_bq_2024_01_06T00_00_00_00_00_5dd91bbda5b6dd71dfa91c649b4b359d
[2024-01-07T22:07:17.889+0000] {taskinstance.py:1400} INFO - Marking task as SUCCESS. dag_id=ingest_from_postgres_to_GCS_bucket, task_id=course_load_dataset_to_bq, execution_date=20240106T000000, start_date=20240107T220712, end_date=20240107T220717
[2024-01-07T22:07:17.953+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 0
[2024-01-07T22:07:18.055+0000] {taskinstance.py:2778} INFO - 0 downstream tasks scheduled from follow-on schedule check
